source/common/stream_info/BUILD

```diff
@@ -27,6 +27,15 @@ envoy_cc_library(
     ],
 )
 
+envoy_cc_library(
```

Comment:
this should be under tcp_proxy, see what is done for PerConnectionCluster, also this should be named override_server_name, "forward" what the filter does and this key only intended for override.

Reply:
My intention was as follows:
The filter is not related directly to the TCP proxy. The filter specifies the intention that the SNI must be forwarded. Other filters, such as TCP proxy, can extract that information and forward the SNI.

So there are three different names here:
1.  _Forward Requested Server Name_ in the Stream Info, means that there is some requested server name, not necessarrily SNI, maybe requested server name from some other protocol, that must be forwarded to the upstream, not necessarily by tcp_proxy.

2. _Forward Original SNI_ - the name of the filter, specifies the intention to forward the SNI, a specific kind of a server name

3. _Override Server Name_ in tcp_proxy, cluster manager etc. it means that the server name of the upstream cluster must be overridden by that value. It does not specify that it is SNI and does not specify where this server name comes from.

This is why I put Forward Requested Server Name in the stream info and not in TCP proxy, to make this artifact independent from the TCP proxy.